src/namespaces/anvil.rs

```diff
@@ -8,6 +8,30 @@ use zksync_types::{Address, H256, U256, U64};
 
 #[rpc]
 pub trait AnvilNamespaceT {
+    /// Create a buffer that represents all state on the chain, which can be loaded to separate
+    /// process by calling `anvil_loadState`.
+    ///
+    /// # Arguments
+    ///
+    /// * `preserve_historical_states` - Whether to preserve historical states
+    ///
+    /// # Returns
+    /// Buffer representing the chain state.
+    #[rpc(name = "anvil_dumpState")]
+    fn dump_state(&self, preserve_historical_states: Option<bool>) -> RpcResult<Bytes>;
```

Comment:
How does it interact with forked networks? I assume that `load_state` will not work on forked network because the state is not empty (or will it?), but if that's the case should we support `dump_state` at all?

Reply:
It will work assuming you are using the same fork for both instances. This is actually a pretty important use case as this feature will be used for CLI options to automatically dump/load state in intervals of time (see anvil's `--dump-state/--load-state/--state/--state-interval`) and we shouldn't prohibit using forks with it.

That being said I did not design any fail-safe mechanism for using a different fork. Writing the fork state *could* be an option but only the user would know if two forks are really the same (URLs can change).